spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -0,0 +1,316 @@
+            1. Let _key_ be _entry_.[[Key]]
+            1. If ! SameValue(_key_, _P_), return *true*.
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-recordget-r-p" aoid="RecordGet">
+        <h1>RecordGet ( _R_, _P_ )</h1>
+        <p>This abstract operation RecordGet takes arguments _R_ and _P_. It performs the following steps when called:</p>
+        <emu-alg>
+          1. Assert: _R_ is an Object that has [[RecordData]] internal slot.
+          1. Assert: Type(_P_) is String.
+          1. Let _rec_ be _R_.[[RecordData]].
+          1. Assert Type(_rec_) is Record.
+          1. Let _len_ be the length of _rec_.
+          1. For each _entry_ of _rec_.[[Fields]], do:
+            1. Let _key_ be _entry_.[[Key]].
+            1. Let _val_ be _entry_.[[Value]].
+            1. Assert: Type(_key_) is String.
+            1. Assert: Type(_val_) is not Object.
+            1. Return _val_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-tuple-exotic-objects">
+      <h1>Tuple Exotic Objects</h1>
+      <p>A Tuple object is an exotic object that encapsuates a Tuple value and exposes virtual integer-indexed data properties corresponding to the individual entries set on the underlying Tuple value. All keys properties are non-writable and non-configurable.</p>
+
+      <p>An object is a <dfn id="tuple-exotic-object">Tuple exotic object</dfn> (or simply, a Record object) if its following internal methods use the following implementations.</p>
+
+      <p>Tuple exotic objects have the same internal slots as ordinary objects. They also have a [[TupleData]] internal slot.</p>
+
+      
+      <emu-clause id="sec-tuple-exotic-objects-setprototypeof-v">
+        <h1>[[SetPrototypeOf]] ( _V_ )</h1>
+        <p>When the [[SetPrototypeOf]] internal method of a Tuple exotic object _T_ is called with argument _V_, the following steps are taken:</p>
+        <emu-alg>
+          1. Return ! SetImmutablePrototype(_T, _V_).
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-tuple-exotic-objects-isextensible">
+        <h1>[[IsExtensible]] ()</h1>
+        <p>When the [[IsExtensible]] internal method of a Tuple exotic object _T_ is called, the following steps are taken:</p>
+        <emu-alg>
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-tuple-exotic-objects-getownproperty-p">
+        <h1>[[GetOwnProperty]] ( _P_ )</h1>
+        <p>When the [[GetOwnProperty]] internal method of a Tuple exotic object _T_ is called with property key _P_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If Type(_P_) is String, then
+            1. Let _numericIndex_ be ! CanonicalNumericIndexString(_P_).
+            1. If _numericIndex_ is not *undefined*, then
+              1. Let _value_ be ? TupleGet(_T_, _numericIndex_)
```

Comment:
Again, you should make sure that TupleGet distinguishes undefined from missing values, as Tuples may contain undefined.